### ISSUE TITLE
Cannot add m3u or Xstream

### ISSUE THREAD

**Issue by arnolddej** [linuxmint/hypnotix#189]
I cannot add a m3u url or Xstream. It shows only the default streams.

**Comment by Axel-Erfurt** [comment]
How did you add it?

**Comment by staeiou** [comment]
Can confirm. Added with the "+ Add a new provider." When I run from the terminal, I see it load the provider I added, but it doesn't appear in the list. 

Free-IPTV: Getting playlist...
Free-IPTV: Checking playlist...
Free-IPTV: Loading channels...
ky: Getting playlist...
ky: Checking playlist...

Platform: Ubuntu 20.04.3 LTS

**Comment by sbadux** [comment]
Sorry to jump in.
Are you adding a specific channel stream (e.g http://sb.err.ee/live/etv.m3u8) or a collection of streams (e.g. https://raw.githubusercontent.com/Free-TV/IPTV/master/playlist.m3u8)?

**Comment by bambinounos** [comment]
Hi guys.
I installed it in Ubuntu 20.04, I have the same problem.
I am trying to add a collection of streams.
By deafult it does not shows any channel.
![imagen](https://user-images.githubusercontent.com/65610134/152913882-b78bb4bd-f75d-4bf9-b591-4e93b70be213.png)
I have been trying to add providers but it only says Downloading playlist and it shows an empty page
![imagen](https://user-images.githubusercontent.com/65610134/152914155-e075f841-90a1-4bc4-9230-69231ae590af.png)
Any idea how to solve it, I would be very thankful.

**Comment by bambinounos** [comment]
> Can confirm. Added with the "+ Add a new provider." When I run from the terminal, I see it load the provider I added, but it doesn't appear in the list.
> 
> Free-IPTV: Getting playlist... Free-IPTV: Checking playlist... Free-IPTV: Loading channels... ky: Getting playlist... ky: Checking playlist...
> 
> Platform: Ubuntu 20.04.3 LTS

Hi, could you fix it?

**Comment by Axel-Erfurt** [comment]
Show the playlist or some lines of it, maybe it is not formatted well.

**Comment by bambinounos** [comment]
Thanks for your reply 

> 

I have tried all options. 
-Stream API (Address, User and paswaord) It does not work
-*.m3u file, http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts

It even does not work with Free-IPTV from Github

**Comment by Axel-Erfurt** [comment]
Can you play that link in terminal with mpv?

`mpv http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts`

**Comment by bambinounos** [comment]
> Can you play that link in terminal with mpv?
> 
> `mpv http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts`

Yes, it works with mpv. But it's not friendly when you need to change channels. But it works

**Comment by Axel-Erfurt** [comment]
Theoretically, it should then also work in hypnotix.

You may be able to play the playlist in vlc.

**Comment by bambinounos** [comment]
Hypnotix does not load the playlist, it just show Downloading playlist and it shows an empty page like captures

**Comment by Axel-Erfurt** [comment]
I've created a test.m3u

```
#EXTM3U
#EXTINF:1,Teststream 1
http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts
#EXTINF:2,Teststream 2
http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts
#EXTINF:3,Teststream 3
http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts
#EXTINF:4,Teststream 4
http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts
```
![screenshot_w](https://user-images.githubusercontent.com/2623542/153216967-69312618-8754-4571-8953-627e4cf76ad3.png)

**Comment by bambinounos** [comment]
> #EXTM3U
> #EXTINF:1,Teststream 1

It works when I added..
Example
#EXTM3U
#EXTINF:1,Teststream 1
but it load just 1 channel.
is there anyway to load all guide at once?

**Comment by bambinounos** [comment]
> I've created a test.m3u
> 
> ```
> #EXTM3U
> #EXTINF:1,Teststream 1
> http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts
> #EXTINF:2,Teststream 2
> http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts
> #EXTINF:3,Teststream 3
> http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts
> #EXTINF:4,Teststream 4
> http://51.222.43.170:36995/get.php?username=User&password=pass&type=m3u&output=ts
> ```
> 
> ![screenshot_w](https://user-images.githubusercontent.com/2623542/153216967-69312618-8754-4571-8953-627e4cf76ad3.png)

Why does not it works?
![imagen](https://user-images.githubusercontent.com/65610134/153227662-7a70ce61-92c0-42cd-83ec-e6951e5487fa.png)
If we add the address, User and Password. It should add all channels ans media.
But it just returns to the previous page.

**Comment by Axel-Erfurt** [comment]
Ok if it works, but I've made this long ago.

The newest version 2.6 you can find at http://packages.linuxmint.com/pool/main/h/hypnotix/

**Comment by bambinounos** [comment]
> Ok if it works, but I've made this long ago.
> 
> The newest version 2.6 you can find at http://packages.linuxmint.com/pool/main/h/hypnotix/

Hi Axel, this version works, but it doesn't load properly. It sometimes gets frozen or loading slow.
